### PR TITLE
Serve requests concurrently

### PR DESCRIPTION
**Pull request by rgrinberg**
Change the return type for `on_request` to allow concurrent handling.

### CODE REVIEW

**Review by mnxn — APPROVED** [review]

**Comment by ulugbekna** [comment]
(reviewing please don't merge yet)

**Comment by rgrinberg** [comment]
Take your time. I removed all the unrelated commits to make it easier to review.

**Review by ulugbekna — APPROVED** [review]
Looks good to me.

File: jsonrpc-fiber/src/jsonrpc_fiber.ml [hunk 1]
```diff
@@ -53,28 +88,30 @@ struct
...
-      let error =
-        Response.Error.make ~code:InternalError ~data
-          ~message:"uncaught exception" ()
-      in
-      Response.error id error
+    | Ok x -> Ok x
+    | Error exns -> Error (response_error_of_exns id exns)
 
-  let on_request_fail ctx : (Response.t * _) Fiber.t =
+  let on_request_fail ctx : (Reply.t * _) Fiber.t =
```
[the remark this answers is not in the document]
Comment (rgrinberg) [reply jsonrpc-fiber/src/jsonrpc_fiber.ml:108]: The way I see it, we can either name something for what it does, or what its purpose is. I prefer the former usually.

**Pull request merged**